src/crm.ts

```diff
@@ -170,10 +168,6 @@ export abstract class CRM implements CRMInterface {
     throw new Error("you need to implement handleContact in your CRM");
   };
 
-  handleMessage = (message: ActionMessageV2 | EventMessageV2): Promise<handleResult | boolean> => {
```

Comment:
it should stay there, and handle our basic doi case (ie call handleEmailChange or call handleContact directly)

Reply:
the "high level" plan is that crm does the default routing of the data flow and based on the type of CRM, calls the right "crm specific" call.

ie handleContact (call crm api to add or update a contact and puts it into the right group/tag/segment) or fetchingCampaign information (so we know how to map proca campaigns/widget to the whatever segmentation tool needs to be used in the CRM)


check handleActionContact for example: based on the type of crm and the event, it knows if it needs to push the contact to the crm. how to push the contact needs to be overridden in the crm, but looking at the event can stay in the generic handleActionContact, unless your crm does something weird and can override it, but in general it should only have to implement handleContact

We need to do the same for the handleMessage for the double opt-in, isn't it?

Reply:
The CRM routing works without this function. I added handleMessage thinking it would simplify handling events (handleEvent) and action messages (handleAction), but it wasn't such an improvement so I removed it now.  It was only ever used for cleverreach.ts (not anymore)